qcodes/instrument_drivers/agilent/E8527D.py

```diff
@@ -10,26 +11,86 @@
 class Agilent_E8527D(VisaInstrument):
     """
     This is the QCoDeS driver for the Agilent_E8527D signal generator.
-
     This driver will most likely work for multiple Agilent sources.
-
     This driver does not contain all commands available for the E8527D but
     only the ones most commonly used.
     """
     def __init__(self, name: str, address: str,
-                 step_attenuator: bool = False,
+                 step_attenuator: Optional[bool] = None,
                  terminator: str = '\n',
                  **kwargs: Any) -> None:
         super().__init__(name, address, terminator=terminator, **kwargs)
 
+        if step_attenuator is not None:
+            warnings.warn("step_attenuator argument to E8527D is deprecated "
+                          "and has no effect. It will be removed in the "
+                          "future.")
+
+        # Query installed options
+        self._options = self.ask_raw('DIAG:CPU:INFO:OPT:DET?')
+
+        # Determine installed frequency option
+        frequency_option = None
+        for f_option in ['513', '520', '521', '532', '540', '550', '567']:
+            if f_option in self._options:
+                frequency_option = f_option
+        if frequency_option == None:
```

Comment:
It is considered better python style to use is rather than == when comparing against None, True or False